src/app/files/fileManager.js

```diff
@@ -42,6 +42,212 @@ class FileManager extends Plugin {
     this.init()
   }
 
+    /**
+   * Emit error if path doesn't exist
+   * @param {string} path path of the file/directory
+   * @param {string} message message to display if path doesn't exist.
+   */
+  _handleExists (path, message) {
+    if (!this.exists(path)) {
+      this._handleError({ code: 'ENOENT', message })
+    }
+  }
+
+  /**
+   * Emit error if path is not a file
+   * @param {string} path path of the file/directory
+   * @param {string} message message to display if path is not a file.
+   */
+  _handleIsFile (path, message) {
+    if (!this.isFile(path)) {
+      this._handleError({ code: 'EISDIR', message })
+    }
+  }
+
+  /**
+   * Emit error if path is not a directory
+   * @param {string} path path of the file/directory
+   * @param {string} message message to display if path is not a directory.
+   */
+  _handleIsDir (path, message) {
+    if (this.isFile(path)) {
+      throw new Error({ code: 'ENOTDIR', message })
+    }
+  }
+
+  /**
+   * Emits error based on error code
+   * @param {object} error error { code, message }
+   */
+  _handleError (error) {
```

Comment:
what;s the reason of having this?

Reply:
So that errors have a uniform format and it is not just thrown everywhere in our codebase.

Reply:
sounds ok to me, what are the concern about that?

Reply:
but it adds nothing more to the throw

Reply:
You're doing three time the job.
For example if the `mkdir` method throw, the message will be: 
"File already exists Cannot create directory ${path}" with the first part beeing handle by the code & the second by the error. But still you have a method `_handleExists`.

Either you have one method to handle all error with a mapping on the error code : 
```typescript
const errorMsg = {
  ENOENT: 'No such file or directory',
  ...
}
const createError = (err) => new Error(`${errorMsg[err]} ${err.message || ''}`)
...
mkdir (path) {
  if (this.exists(path)) {
    throw createError({ code: 'EEXIST', message: `Cannot create directory ${path}` })
  }
}
```
Or you have a named methods that take a string as argument : 
```typescript
_handleEnoent(msg) {
  throw new Error('No such file or directory ' + msg);
}
```
Or you can use assert methods by merging check & error : 
```typescript
_assertExist(path, msg) {
  if (....) {
    throw new Error('No such file or directory ' + msg);
  }
}
```